test_conformance/extensions/cl_khr_command_buffer/command_buffer_get_command_buffer_info.cpp

```diff
@@ -0,0 +1,340 @@
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RecordCommandBuffer()
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+#define test_expected_info(cond)                                               \
+    {                                                                          \
+        if (cond)                                                              \
+        {                                                                      \
+            log_error("clGetCommandBufferInfoKHR return values not as "        \
+                      "expected\n");                                           \
+            return TEST_FAIL;                                                  \
+        }                                                                      \
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunQueuesInfoTest()
+    {
+        cl_int error = TEST_PASS;
+
+        // record command buffers
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        // vector containter added due to potential future growth, at the moment
+        // spec of cl_khr_command_buffer says command-buffer accepts only 1
+        // queue
+        std::vector<cl_command_queue> expect_queue_list = { queue };
+        cl_uint num_queues = 0;
+        size_t ret_value_size = 0;
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_NUM_QUEUES_KHR, sizeof(cl_uint),
+            &num_queues, &ret_value_size);
+        test_error(error, "clGetCommandBufferInfoKHR failed");
+
+        test_expected_info(ret_value_size > sizeof(cl_int));
+
+        test_expected_info(num_queues != expect_queue_list.size());
+
+        std::vector<cl_command_queue> queue_list(num_queues);
+        size_t expect_size = queue_list.size() * sizeof(cl_command_queue);
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_QUEUES_KHR, expect_size,
+            &queue_list.front(), &ret_value_size);
+        test_error(error, "clGetCommandBufferInfoKHR failed");
+
+        test_expected_info(ret_value_size > expect_size);
+
+        // We can not check if this is the right queue because this is an opaque
+        // object, test against NULL.
+        for (int i = 0; i < queue_list.size(); i++)
+        {
+            test_expected_info(queue_list[i] == NULL);
+        }
```

Comment:
I think we should be able to do a little better here, see: https://github.com/KhronosGroup/OpenCL-Docs/issues/797

Reply:
Done, confusion comes from two directions:

1) similar solution used in a different test (raise an issue?):
[/test_conformance/events/test_events.cpp](https://github.com/KhronosGroup/OpenCL-CTS/blob/296a7a574dc09e8b7c143e1de77ed5ae0aa8b63d/test_conformance/events/test_events.cpp#L143-L151)
```
    // We can not check if this is the right queue because this is an opaque
    // object.
    if (size != sizeof(queue))
    {
        log_error("ERROR: Returned command queue size does not validate "
                  "(expected %d, got %d)\n",
                  (int)sizeof(queue), (int)size);
        return -1;
    }
```
2) PoCL fails equality test at the moment